### ISSUE TITLE
build error

### ISSUE THREAD

**Issue by ravener** [fmang/oshu#95]
when building I got this:
```
[  9%] Building CXX object lib/CMakeFiles/liboshu.dir/audio/stream.cc.o
/home/ravener/repos/oshu/lib/audio/stream.cc: In function ‘int open_demuxer(const char*, oshu::stream*)’:
/home/ravener/repos/oshu/lib/audio/stream.cc:196:17: error: invalid conversion from ‘const AVCodec**’ to ‘AVCodec**’ [-fpermissive]
  196 |                 (const AVCodec**)&stream->codec,
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                 |
      |                 const AVCodec**
In file included from /home/ravener/repos/oshu/lib/audio/stream.cc:10:
/usr/include/x86_64-linux-gnu/libavformat/avformat.h:2301:35: note:   initializing argument 5 of ‘int av_find_best_stream(AVFormatContext*, AVMediaType, int, int, AVCodec**, int)’
 2301 |                         AVCodec **decoder_ret,
      |                         ~~~~~~~~~~^~~~~~~~~~~
make[2]: *** [lib/CMakeFiles/liboshu.dir/build.make:118: lib/CMakeFiles/liboshu.dir/audio/stream.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:256: lib/CMakeFiles/liboshu.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```
I fixed this by removing the `const` keyword and everything built fine.

I would've opened a PR but first wanted to see what's going on since no one else seems to have this issue. but if removing the keyword won't hurt others it may improve compatibility for some.

**Comment by ravener** [comment]
oh i see #93 now, well seems like Ubuntu is still on ffmpeg 4 unfortunately, this makes it difficult to build with distro libraries. is it possible to check that version via macros and conditionally support both versions?

**Comment by fmang** [comment]
Thanks for your investigation. I pushed the branch ffmpeg_4 which adds a version check. Please tell me if that fixes your issue.

**Comment by ravener** [comment]
@fmang I just tried and it works fine

**Comment by fmang** [comment]
Fixed by d381257f5695e26b03041021d2e698747d7c1f1e.
